### ISSUE TITLE
Mishandled serialization on some Exception types with non-serializable fields

### ISSUE THREAD

**Issue by maryamariyan** [dotnet/corefx#35041]
Not all Exception types are serializable but for those (listed below) they serialize a private field of arbitrary type that may not be serializable:

- InvalidOperationException
- ArgumentOutOfRangeException
- RuntimeWrappedException 
- XmlSchemaValidationException

For them, we need to use the ToString() value for the non serializable types. (sample to follow: SwitchExpressionException)

Refer to: https://github.com/dotnet/corefx/pull/34954#discussion_r252381080

cc: @danmosemsft

**Comment by jkotas** [comment]
This is one of the "by design" problems of binary serialization. I think this would be hard to fix in a compatible way.

`Exception.Data` or all collections have the same problem. If you happen to point them to non-serializable object, the whole thing will fail to serialize.

**Comment by danmosemsft** [comment]
I agree that we can't solve it for collections. For exceptions, it may be less deterministic what type gets stuffed into them, and a path that's not well tested. It's also a case where something (a string) may well have non zero value. 
In the case of XML Exception, I would certainly appreciate a ToString on the XML node over a new exception while trying to transmit my exception. The change is also small and safe.
